internal/daemon/controller/common/scopeids/scope_ids.go

```diff
@@ -353,3 +353,143 @@ func GetListingScopeIds(
 	}
 	return scopeResourceInfo.ScopeIds, scopeMap, nil
 }
+
+// AuthorizedForList retrieves and returns all scopes where a user is authorized
+// to perform a *list* action on. It looks recursively from `rootScopeId`.
+func AuthorizedForList(
```

Comment:
Adding a note for us to follow up with some unit tests for this prior to merging the llb

Reply:
After merge here into the LLB, I'll link these comments over 👍